src/Native/Unix/configure.cmake

```diff
@@ -407,6 +407,18 @@ check_function_exists(
     getpeereid
     HAVE_GETPEEREID)
 
+check_function_exists(
+    getdomainname
+    HAVE_GETDOMAINNAME)
+
+check_function_exists(
+    uname
+    HAVE_UNAME)
+
+if(NOT HAVE_GETDOMAINNAME AND NOT HAVE_UNAME)
```

Comment:
does all Linux distros support one of them? or we can run into a disto have other alternatives?